docs/design_proposals/sync-improvements.md

```diff
@@ -102,7 +102,7 @@ type Builder interface {
    DependenciesForArtifact(ctx context.Context, artifact *latest.Artifact) ([]string, error)
 
    // ADDED
-   SyncMap() (map[string][]string, error)
+   SyncMap(ctx context.Context, artifact *latest.Artifact) (map[string][]string, error)
```

Comment:
should this return a `syncMap`? (Not sure it's better)

Reply:
@balopat had the same idea in `pkg/skaffold/sync/sync.go`. I'm also very unsure what is better. My main issue is that a type alias binds the using sites tighter to the place where `type SyncMap` is defined (probably `builder.go`?). Besides, I don't particularly like that there will be two `SyncMap` symbols in `builder.go`: 1 the function, 2 its return type.
Overall me feeling is slightly towards keeping it as is. The local type alias in `sync.go` makes somewhat sense, even though it cannot substitute all usages of `map[string][]string` due to visibility.